lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java

```diff
@@ -90,7 +90,7 @@ public static software.amazon.awssdk.core.retry.RetryMode getRetryMode(RetryMode
     private String sseKmsKeyId;
     private DataSize streamingPartSize = DataSize.of(16, MEGABYTE);
     private boolean requesterPays;
-    private Integer maxConnections;
+    private Integer maxConnections = 500;
```

Comment:
Let's have a comment why this number. (i guess the reason is more than just "it's what we used to have in the legacy", otherwise we wouldn't make this change -- there are other things we used to have and we wanted not to carry on)